Examples/VaporExample/README.md

```diff
@@ -1,13 +1,70 @@
+1. A DELETE request  at the URL `/kittens/{name}` deletes the kitten with the specified name.
+
+### MongoDB Usage
+This application connects to a local standalone MongoDB server running on the default host/port, `localhost:27017`. It uses the collection "kittens" in the database "home". The "kittens" collection has a [unique index](https://docs.mongodb.com/manual/core/index-unique/) on the "name" field, ensuring that no two kittens in the collection can have the same name.
+
+If you'd like to point the application to a MongoDB server elsewhere (e.g. on [MongoDB Atlas](https://www.mongodb.com/cloud/atlas)) or running on a different port, or change any configuration options for the client, you can edit the call to `app.mongoDB.configure()` in `Sources/App/configure.swift`.
+
+The call to `configure()` initializes a global `MongoClient` to back your application. `MongoClient` is implemented with that approach in mind: it is safe to use across threads, and is backed by a [connection pool](https://en.wikipedia.org/wiki/Connection_pool) which enables sharing resources throughout the application.
+
+Throughout your application, you can access the global client via `app.mongoDB.client`. Note that the global client may return `EventLoopFuture`s on *any* `EventLoop` in the application's `EventLoopGroup`, so if you use this client you will need to ensure you "hop" the futures back to the event loop you are currently on. See the `EventLoopFuture` [documentation](https://apple.github.io/swift-nio/docs/current/NIO/Classes/EventLoopFuture.html) for more details.
+
+To avoid the need to hop `EventLoop`s, whenever you are using MongoDB in a request handler, we strongly recommend you use an `EventLoopBoundMongoClient` instead, accessible via `req.mongoDB.client`. This type is a small wrapper around the global client, which returns `EventLoopFuture`s on a specific `EventLoop` which it is "bound" to. Using an `EventLoopBoundMongoClient` that is backed by the same `EventLoop` as a `Request` means you can use the client within a request handler without worrying about thread safety. You can access an `EventLoopBoundMongoClient` for a `Request` via `req.mongoDB.client`.
+
+`MongoDatabase`s and `MongoCollection`s you retrieve from an `EventLoopBoundMongoClient` will automatically be bound to the same event loop as the parent client.
+
+For convenience, we recommend adding your own computed properties to `Request` that return `MongoDatabase`s and `MongoCollection`s you frequently access, as is shown in `Sources/App/Kitten.swift` 
+with the `kittenCollection` property:
+```swift
+extension Request {
+    /// Convenience extension for obtaining a collection which uses the same event loop as a request.
+    var kittenCollection: MongoCollection<Kitten> {
+        self.mongoDB.client.db("home").collection("kittens", withType: Kitten.self)
+    }
+}
+```
+### Codable Usage
```

Comment:
one thing to note is that our existing example doesn't make use of the ExtendedJSONEncoder/ExtendedJSONDecoder as ContentEncoder/ContentDecoder feature. 

It seems good to demo it though it's not something all users actually need to do, e.g. this app works fine without it, so I wasn't totally sure whether to do it. We could make this app need the feature by adding some field to `Kitten` that is not serializable by `JSONEncoder` if we want.

Reply:
Yeah I think that's a good idea. I think it's important we indicate these encoders / decoders exist, what they're used for, and how to enable them (both in example and readme), since otherwise I imagine most users will just not notice or use them at all.